flink-end-to-end-tests/test-scripts/test-runner-common.sh

```diff
@@ -58,23 +59,28 @@ function run_test {
     exit_code="$?"
     # remove trap for test execution
     trap - ERR
-    post_test_validation ${exit_code} "$description" "$skip_check_exceptions"
+    post_test_validation ${exit_code} "$description" "$check_exceptions_action" "$check_command"
 }
 
 # Validates the test result and exit code after its execution.
 function post_test_validation {
     local exit_code="$1"
     local description="$2"
-    local skip_check_exceptions="$3"
+    local check_exceptions_action="$3"
 
     local time_elapsed=$(end_timer)
 
-    if [[ "${skip_check_exceptions}" != "skip_check_exceptions" ]]; then
+    if [[ "${check_exceptions_action}" == "skip_check_exceptions" ]]; then
+        echo "Checking of logs skipped."
+    elif [[ "${check_exceptions_action}" == "custom_check_exceptions" ]]; then
+        echo "Custom exception checking of logs."
+        local check_command="$4"
```

Comment:
It's better to define it at the beginning of the method to clearly show all the params.